### ISSUE TITLE
How to distribute training data and split them into workers instead of loading all data in all workers 

### ISSUE THREAD

**Issue by WenkaiBai** [linkedin/TonY#144]
In demo, all training data are on all workers, 
My situation is that training data is too big to store on single worker.

How to distribute data?

**Comment by oliverhu** [comment]
Good question. TonY is the layer to drive the distributed training, in terms of how to split the data, it is up the training code. A simplified way would be: since we have the NUM_WORKERS, TASK_INDEX environment set up for each container's environment, you can get your portion of the data base on that. For example: you have 10 data files and 10 containers, each container can read its portion by mapping its index to the data file.

**Comment by erwa** [comment]
Hi @WenkaiBai,

You shouldn't need to store the full data on each worker; you should be able to stream it in to each worker. Distributed TF is designed so every worker reads the full dataset: see this SO post: https://stackoverflow.com/a/45994422/1128392. However, you can shuffle your dataset to apply some randomization so that each worker trains on a different perturbed version of the dataset.

**Comment by WenkaiBai** [comment]
Thank you all for answering. now I have a clue how to distribute my data. Thanks!!!!!
